frontend/gateway/gateway.go

```diff
@@ -310,29 +349,51 @@ func (lbf *llbBridgeForwarder) Solve(ctx context.Context, req *pb.SolveRequest)
 		return nil, err
 	}
 
-	exp := map[string][]byte{}
-	if err := json.Unmarshal(req.ExporterAttr, &exp); err != nil {
-		return nil, err
-	}
+	ids := make(map[string]string)
 
-	if expResp != nil {
-		for k, v := range expResp {
-			exp[k] = v
+	lbf.mu.Lock()
+	for k, ref := range res {
+		id := identity.NewID()
```

Comment:
May be better to have function or closure + `defer lbf.mu.Unlock()` in the case `identity.NewID()` panics.

Reply:
Nevermind, this runs in a goroutine without recover()